aws-predictions-tfl/src/main/java/com/amplifyframework/predictions/service/TFLiteTextClassificationService.java

```diff
@@ -0,0 +1,332 @@
+
+/**
+ * An implementation of text classification service using
+ * pre-trained model from Tensorflow lite.
+ */
+final class TFLiteTextClassificationService {
+    private static final String SERVICE_KEY = "textClassifier";
+    private static final String MODEL_PATH = "text_classification.tflite";
+    private static final String DICTIONARY_PATH = "text_classification_vocab.txt";
+    private static final String LABEL_PATH = "text_classification_labels.txt";
+
+    // The maximum length of an input sentence.
+    private static final int SENTENCE_LEN = 256;
+
+    // Percentage multiplier
+    private static final int PERCENT = 100;
+
+    // Simple delimiter to split words.
+    private static final String SIMPLE_SPACE_OR_PUNCTUATION = "[ ,.!?\n]";
+
+    /*
+     * Reserved values in ImdbDataSet dictionary:
+     * dictionary["<PAD>"] = 0      used for padding
+     * dictionary["<START>"] = 1    mark for the start of a sentence
+     * dictionary["<UNKNOWN>"] = 2  mark for unknown words (OOV)
+     */
+    private static final String PAD = "<PAD>";
+    private static final String START = "<START>";
+    private static final String UNKNOWN = "<UNKNOWN>";
+
+    private final Map<String, Integer> dictionary = new HashMap<>();
+    private final List<String> labels = new ArrayList<>();
+
+    private final Context context;
+
+    private Interpreter tflite;
+    private AtomicBoolean loaded;
+
+    /**
+     * Constructs an instance of service to perform text
+     * sentiment interpretation using Tensorflow Lite
+     * interpreter.
+     * @param context the Android context for loading model
+     */
+    TFLiteTextClassificationService(Context context) {
+        this.context = context;
+        this.loaded = new AtomicBoolean(false);
+
+        try {
+            // Try loading assets now if possible
+            loadIfNotLoaded();
+        } catch (PredictionsException exception) {
+            // Ignore if it fails to load during configuration.
+
+            // This may sound weird, but we want the model to be loaded in as soon as possible.
+            // But we don't want to throw an error that needs to be caught at configuration time,
+            // since these models are supposed to be optional.
+        }
+    }
+
+    /**
+     * Gets the associated service key of this service for
+     * identification.
+     * @return the service key
+     */
+    @NonNull
+    String getServiceKey() {
+        return SERVICE_KEY;
+    }
+
+    /**
+     * Classifies text to analyze associated sentiments.
+     * @param text the text to classify
+     * @param onSuccess notified when classification succeeds
+     * @param onError notified when classification fails
+     */
+    void classify(
+            @NonNull String text,
+            @NonNull Consumer<InterpretResult> onSuccess,
+            @NonNull Consumer<PredictionsException> onError
+    ) {
+        final Sentiment sentiment;
+        try {
+            loadIfNotLoaded();
```

Comment:
The model loading in this category is analogous to the initialization work, in the DataStore.

I added some lifecycle to accomodate this kind of stuff.

`configure()` on the plugin reads the config file and instantiates some classes.

`intialize()` can be used for this kind of heavy-lifting stuff, though.

Checkout the `afterInitializion()` method in DataStore. We could externalize this utility into core. Then, both categories could use it.

Right now, the logic is like:
1. Try to load models at start up.
2. Before doing anything, try again.

And its sort of relaxed about sequencing/error handling.

With the new approach, it'd be more like:
1. Try to load model at initialization. If it fails, pass back error.
2. Defer all operations _until after_ that one initialization attempt succeeds/fails.
3. Only ever try to initialize at a single point in time during the operation of the system.

Reply:
the point is to relax the error handling. This is because models are optional for this plugin, which means that this plugin needs to successfully configure itself even if all of the models are missing (and therefore is useless). 

Let's look at this scenario:
- I want to detect sentiment
  - I want an accurate result for this particularly, so I want the result to be union of both offline and online results
- I want to translate text
  - memory is precious, so I'll only use the cloud resource for this
- I set network mode to `auto` so that it will combine result if both results are available
- If offline mode doesn't deliver (due to missing model or something), then only online result should be delivered

This use case is achieved by only downloading the required TFLite model for sentiment detection. Both online plugin and offline plugin needs to configure itself upon `Amplify.configure()`. We can't just crash the offline plugin or even leave it hanging at this point waiting for successful load of every model. The configuration needs to go through so that when the customer decides to:
1. call sentiment detection: both online and offline plugins must return result
2. call translate: online plugin returns result, but offline plugin may fail

My solution was to try loading at initialization but ignore error to allow configuration to succeed. That way, we can actually handle each API and their respective model requirement separately.